src/probnum/diffeq/odesolver.py

```diff
@@ -28,14 +28,20 @@ def solve(self, firststep, steprule, **kwargs):
         """
         t, current_rv = self.initialise()
         times, rvs = [t], [current_rv]
-        stepsize = firststep
+        stepsize = steprule.firststep
 
         while t < self.ivp.tmax:
 
             t_new = t + stepsize
             proposed_rv, errorest = self.step(t, t_new, current_rv, **kwargs)
-
-            if steprule.is_accepted(stepsize, errorest):
+            internal_norm = steprule.errorest_to_internalnorm(
+                errorest=errorest,
+                proposed_state=proposed_rv.mean,
+                current_state=current_rv.mean,
+                atol=atol,
+                rtol=rtol,
+            )
```

Comment:
I understand that making the computation of this internal norm thingy (naming inspired by SciML) part of the steprule is controversial, I added a comment in the code in AdaptiveSteps to explain why

Reply:
Honestly I think in our OOP code it kind of makes sense. But for me it raises the question if "atol" and "rtol" should not be part of the steprule in this case?

Also I never completely understood why it's called "internalnorm" there, so if you have any better idea it would be perfectly fine for me to change that :D

Reply:
> Honestly I think in our OOP code it kind of makes sense. But for me it raises the question if "atol" and "rtol" should not be part of the steprule in this case?

I dont have a strong opinion here. I kind of like that it is not part of the steprule:
I think that at least mathematically, it is rather elegant to have them separately, in the sense that (in PN algorithm language) it decouples action rules from stopping criteria. Also, we can setup a problem once (solver, prior, steprule etc.) and then solve the problem for different tolerances without redefining old objects (e.g. in workprecision diagrams)

Reply:
called "errorest_to_norm" now, and since steprule has attributes atol and rtol, this should indeed be a method